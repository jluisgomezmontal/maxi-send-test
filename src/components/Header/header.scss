.btn_header{
  padding: 5px 15px;
  background-color: rgb(59, 57, 57);
  text-decoration: none;
  border-radius: 3px;
  color: white;
  margin-left: 25px;
  transition: .5s ease;
}

.btn_header:hover{
  background-color: rgb(75, 72, 72);
}
