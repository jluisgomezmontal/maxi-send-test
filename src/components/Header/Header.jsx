import { useLocation, Link } from "react-router-dom";
import "./header.scss";
export const Header = () => {
  let { pathname } = useLocation();

  return (
    <>
      <nav className="navbar  navbar-expand-lg navbar-dark bg-primary">
        <div className="container-xl">
          <Link className="navbar-brand ">
            <h1 className="fw-bold p-0 m-0 my-1">DEMO Streaming</h1>
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarTogglerDemo02"
            aria-controls="navbarTogglerDemo02"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarTogglerDemo02">
            <ul className="navbar-nav me-auto mb-2 mb-lg-0"></ul>
            <form className="form-inline">
              <Link className="text-white text-decoration-none">Log in</Link>
              <Link className="btn_header">Start your free trial</Link>
            </form>
          </div>
        </div>
      </nav>
      <div className="bg-steel text-white">
        <div className="container-xl">
          <h2 className="py-2">
            Popular{" "}
            {pathname === "/series"
              ? "Series"
              : pathname === "/movies"
              ? "Movies"
              : "Titles"}
          </h2>
        </div>
      </div>
    </>
  );
};
