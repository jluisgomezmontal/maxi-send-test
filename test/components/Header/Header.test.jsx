/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { Header } from "../../../src/components/Header/Header";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";

describe("testing <Header/>", () => {
  test("should render propertly", () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("DEMO Streaming")).toBeInTheDocument();
    expect(screen.getByText("Log in")).toBeInTheDocument();
    expect(screen.getByText("Start your free trial")).toBeInTheDocument();
  });
});
