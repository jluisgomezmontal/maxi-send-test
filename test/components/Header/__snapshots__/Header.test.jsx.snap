// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <Header/> should render propertly 1`] = `
<div>
  <nav
    class="navbar navbar-dark bg-primary justify-content-between"
  >
    <div
      class="container-xl"
    >
      <a
        class="navbar-brand "
        href="/"
      >
        <h1
          class="fw-bold p-0 m-0 my-1"
        >
          DEMO Streaming
        </h1>
      </a>
      <form
        class="form-inline"
      >
        <a
          class="text-white text-decoration-none"
          href="/"
        >
          Log in
        </a>
        <a
          class="btn_header"
          href="/"
        >
          Start your free trial
        </a>
      </form>
    </div>
  </nav>
  <div
    class="bg-steel text-white"
  >
    <div
      class="container-xl"
    >
      <h2
        class="py-2"
      >
        Popular
         
        Titles
      </h2>
    </div>
  </div>
</div>
`;
